third_party/terraform/resources/resource_storage_bucket.go

```diff
@@ -354,9 +403,25 @@ func resourceStorageBucketCreate(d *schema.ResourceData, meta interface{}) error
 		return err
 	}
 
+	d.SetId(res.Id)
+
+	if v, ok := d.GetOk("retention_policy"); ok {
+		retention_policies := v.([]interface{})
+
+		sb.RetentionPolicy = &storage.BucketRetentionPolicy{}
+
+		retentionPolicy := retention_policies[0].(map[string]interface{})
+
+		if locked, ok := retentionPolicy["is_locked"]; ok && locked.(bool) {
+			err = lockRetentionPolicy(config.clientStorage.Buckets, bucket, res.Metageneration)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
 	log.Printf("[DEBUG] Created bucket %v at location %v\n\n", res.Name, res.SelfLink)
```

Comment:
This log message is probably worth moving above the `SetId` call after creation, and we can log that a retention policy lock was enabled here instead.